diff -u3 Crypt-SSLeay-0.57_01.orig/Makefile.PL Crypt-SSLeay-0.57_01/Makefile.PL
--- Crypt-SSLeay-0.57_01.orig/Makefile.PL	2008-02-18 06:39:06.000000000 -0700
+++ Crypt-SSLeay-0.57_01/Makefile.PL	2010-05-15 11:20:08.408474400 -0600
@@ -19,6 +19,7 @@
 );
 
 $opt_default ||= $ENV{CRYPT_SSLEAY_DEFAULT};
+$opt_libpath ||= $ENV{OPENSSL_PREFIX};
 
 # FIND POSSIBLE SSL INSTALLATIONS
 
@@ -342,11 +343,11 @@
     my $version;
     my $type;
     while (<VERSION_FILE>) {
-        if (/^#define\s+$version_match\s+0x0+(\d\d\d)/) {
+        if (/^#define\s+$version_match\s+0x(\d\d\d\d\d)/) {
             $version = $1;
-            $version =~ s/(\d)0(\d)/$1$2/;            
+            $version =~ s/(\d)0(\d)0(\d)/$1$2$3/;            
             $type = ($version > 92) ? "OpenSSL" : "SSLeay";
-            $version = join('.', split(//, "0$version"));
+            $version = join('.', split(//, "$version"));
             last;
         }
     }